scripts/test-ci-all.sh

```diff
@@ -23,9 +23,13 @@ fi
 # Note: Respect 'CARGO_PROFILE' that crane uses
 cargo build ${CARGO_PROFILE:+--profile ${CARGO_PROFILE}} --all --all-targets
 
+function dump_db_content_on_termination() {
+  trap 'if [ "$?" -ne 0 ]; then echo "##Client Database"; dbtool $FM_DATA_DIR/client.db dump $FM_DATA_DIR clientpass client; echo "##Server Database"; dbtool $FM_DATA_DIR/server-0/database dump -- $FM_DATA_DIR/server-0 pass0; fi' EXIT
+}
+
 function cli_test_reconnect() {
   set -eo pipefail # pipefail must be set manually again
-  trap 'echo "## FAILED: ${FUNCNAME[0]}"' ERR 
+  trap 'dump_db_on_failure; echo "## FAILED: ${FUNCNAME[0]}"' ERR 
```

Comment:
`dump_db_on_failure` vs `dump_db_content_on_termination`? Also you don't need to trap here. `dump_db_content_on_termination` already does. 

For this reason I would consider renaming to `register_trap_dump_db_content_on_failure` or something.

Also you need to export the bash function to call it from these (also exported) test functions. See  e.g. `export -f cli_test_upgrade`

Reply:
Should work, but make sure to try it out, by introducing some `exit 1` in one of the test scripts somewhere (after the federation db already exist).

Reply:
> `dump_db_on_failure` vs `dump_db_content_on_termination`? Also you don't need to trap here. `dump_db_content_on_termination` already does.
> 
So I need to just call this function in each test function.